app/docs/dev.md

```diff
@@ -1,29 +1,207 @@
-  }
-}
-```
-
-```python
-{
-  "example": {
-    "Something": "something"
-  }
-}
-```
\ No newline at end of file
+
+Implemented versions of these examples can be found in the module 'Examples: Evaluation Functions'.
+
+### 1 Checking if a set of quantities match the Buckingham pi theorem
+
+#### a)
+
+In this example the task is: Given $U$, $L$ and $\nu$, suggest a dimensionless group.
+
+For this problem we do not need to predefine any quantities and give exact dimensions. The algorithm assumes that all symbols in the answer (that are not numbers or predefined constants such as $\pi$) are quantities and that there are no other quantities that should appear in the answer.
+
+**Note:** This means that the algorithm does not in any way check that the stated answer is dimensionless, that is left to the task author.
+
+For this example an EXPRESSION response area is used with answer set to `U*L/nu`. It is not necessary to use this specific answer, any example of a correct dimensionless group should work.
+
+With default settings it is required to put `*` (or `/`) between each part of the response and answer. To remove this requirement the grading parameter `strict_syntax` is set to false. Since `nu` is a multicharacter symbol it needs to be added as an input symbol.
+
+#### b)
+
+In this example the task is: Given $U$, $L$, $\nu$ and $f$, determine the necessary number of dimensionless groups and give one example of possible expressions for them.
+
+This task is similar to example a) with two significant differences. First, adding $f$ means that there are now two groups required, and second the task will constructed by defining the quantities and let the function compute the rest on its own instead of supplying a reference example.
+
+For this example an EXPRESSION response area is used with `quantities` set to `('U','(length/time)') ('L','(length)') ('nu','(length**2/time)') ('f','(1/time)')` and `answer` set to `-`.
+
+With default settings it is required to put `*` (or `/`) between each part of the response and answer. To remove this requirement the grading parameter `strict_syntax` is set to false. Since `nu` is a multicharacter symbol it needs to be added as an input symbol.
+
+#### c)
+
+In this example the task is:
+Suppose we are studying water waves that move under the influence of gravity. We suppose that the variables of interest are the acceleration in free fall $g$, the velocity of the wave $v$, the height of the wave $h$ and the wave length $\ell$. We also suppose that they are related by a dimensionally consistent equation $f(g,v,h,l) = 0$. Determine the minimum number of dimensionless $\pi$-variables needed to describe this problem according to the Buckingham pi-theorem and give one example of possible expressions for the dimensionless quantities.
+
+For this problem two dimensionless groups are needed, see the worked solution for a terse solution that gives the general form of the dimensionless quantities.
```

Comment:
where is the worked solution?

Reply:
Worked solution added to end of examples